internal/pkg/runtime/engines/imgbuild/create.go

```diff
@@ -121,12 +97,69 @@ func (engine *EngineOperations) CreateContainer(pid int, rpcConn net.Conn) error
 		}
 	}
 
+	dest = filepath.Join(sessionPath, "proc")
```

Comment:
What's the reasoning for having these mounted after `%setup` is run? My understanding of `%setup` is that its supposed to be ran after mounts are in place and just before the `chroot`

Reply:
`%setup` doesn't need `/proc`, `/sys`, `hosts` and `resolv.conf` since not chrooted, but actually running setup after `hosts` and `resolv.conf` mount doesn't allow to inject a custom `/etc/hosts` in image without overriding `/etc/hosts` on system :

```
%files
custom_hosts /etc/hosts

%setup
     cat > $SINGULARITY_ROOTFS/etc/hosts <<EOF
blahblah ....
EOF
```

this patch additionally prevent accidental override, so its required to run `%setup` `%files` before `hosts` and `resolv.conf` mounts to not get a permission denied error.

Reply:
Makes sense